app/resources/api/v2/tube_resource.rb

```diff
@@ -40,10 +40,11 @@ class TubeResource < BaseResource
       filter :barcode, apply: ->(records, value, _options) { records.with_barcode(value) }
       filter :uuid, apply: ->(records, value, _options) { records.with_uuid(value) }
       filter :purpose_name, apply: (lambda do |records, value, _options|
-        purpose = Purpose.find_by(name: value)
+        purpose = Purpose.where(name: value)
         records.where(plate_purpose_id: purpose)
```

Comment:
This adjustment lest us pass in either a single purpose name, or an array of them.